src/modules/Login/action.js

```diff
@@ -32,85 +27,101 @@ export const initializeAccount = (account: AbcAccount, touchIdInfo: Object) => a
+    } else {
+      // We have a wallet
+      const {walletId, currencyCode} = ACCOUNT_API.getFirstActiveWalletInfo(account, currencyCodes)
+      accountInitObject.walletId = walletId
+      accountInitObject.currencyCode = currencyCode
+    }
+    const {activeWalletIds, archivedWalletIds, currencyWallets} = account
+    for (const walletId of Object.keys(currencyWallets)) {
+      const abcWallet:AbcCurrencyWallet = currencyWallets[walletId]
+      if (abcWallet.type === 'wallet:ethereum') {
+        if (state.ui.wallets && state.ui.wallets.byId && state.ui.wallets.byId[walletId]) {
+          const enabledTokens = state.ui.wallets.byId[walletId].enabledTokens
+          abcWallet.enableTokens(enabledTokens)
+        }
       }
-    })
-
-  SETTINGS_API.getLocalSettings(account)
-    .then((settings) => {
-      const localDefaults = SETTINGS_API.LOCAL_ACCOUNT_DEFAULTS
-
-      const localFinal = {...localDefaults, ...settings}
-      // Add all the local settings to UI/Settings
-      dispatch(SETTINGS_ACTIONS.setBluetoothMode(localFinal.bluetoothMode))
-    })
-
-  SETTINGS_API.getCoreSettings(account)
-    .then((settings) => {
-      const coreDefaults = SETTINGS_API.CORE_DEFAULTS
-
-      const coreFinal = {...coreDefaults, ...settings}
-      dispatch(SETTINGS_ACTIONS.setPINMode(coreFinal.pinMode))
-      dispatch(SETTINGS_ACTIONS.setOTPMode(coreFinal.otpMode))
-    })
+    }
+    accountInitObject.activeWalletIds = activeWalletIds
+    accountInitObject.archivedWalletIds = archivedWalletIds
+    accountInitObject.currencyWallets = currencyWallets
+
+    const settings = await SETTINGS_API.getSyncedSettings(account)
+    const syncDefaults = SETTINGS_API.SYNCED_ACCOUNT_DEFAULTS
+    const syncFinal = {...syncDefaults, ...settings}
+    const customTokens = settings ? settings.customTokens : []
+    accountInitObject.autoLogoutTimeInSeconds = syncFinal.autoLogoutTimeInSeconds
+    accountInitObject.defaultFiat = syncFinal.defaultFiat
+    accountInitObject.merchantMode = syncFinal.merchantMode
+    accountInitObject.customTokens = syncFinal.customTokens
+    accountInitObject.denominationKeys.push({currencyCode: 'BTC', denominationKey: syncFinal.BTC.denomination})
+    accountInitObject.denominationKeys.push({currencyCode: 'BCH', denominationKey: syncFinal.BCH.denomination})
+    accountInitObject.denominationKeys.push({currencyCode: 'ETH', denominationKey: syncFinal.ETH.denomination})
+    if (customTokens) {
+      customTokens.forEach((token) => {
+        // dispatch(ADD_TOKEN_ACTIONS.setTokenSettings(token))
+        accountInitObject.customTokensSettings.push(token)
+        // this second dispatch will be redundant if we set 'denomination' property upon customToken creation
+        accountInitObject.denominationKeys.push({currencyCode: token.currencyCode, denominationKey: token.multiplier})
+      })
+    }
+    const localSettings = await SETTINGS_API.getLocalSettings(account)
```

Comment:
Does this function return defaults if it can't load the settings? If so, do we really need to merge the settings on the following lines?